#+title: Discussion

#+SETUPFILE: ~/.doom.d/setup-files/latex.setup
#+LATEX_HEADER: \addbibresource{~/biblio/pop-gen.bib}

When calculating genome-wide levels of polymorphism, we found levels of both \theta_{W} and pi to vary significantly for different regions of the genome. While this is expected due to varying levels of mutations and recombination, selection of certain genes and region is also a plausible explaination. It remains to be indentified if a significant proportion of putative genes under selection coincide with the observed regions of reduced polymorphism found. Furthermore, we found Tajima's D to vary across the genome, but generally being strongly negative for both chromosomes analyzed in each subpopulation separately, as well as for the total population. Negative values of Tajima's D across multiple chromosomes can be sign of population expansion, which would be consistent with findings of other studies ([[cite:&rich-2000-popul-struc]]). This is likely linked to expansion of human populations and increased vectorial capacity since the last glacial maximum ([[cite:&donnelly-2001-eviden-recen]]).

We investigated population structure in /P. falciparum/ samples from the African countries of Gambia, Democratic Republic of Congo, Kenya and Tanzania. Applying DAPC analysis, we found very little population structure, with samples from Gambia being the most differntiated in one of the two chromosomes, supporting a weak form of isolation by  distance, as Gambia is the most distant from the other three countries. It is feasible, that this relative lack of population structure is due to strong migration from humans, as human movement has shown to influence parasite movement at smaller geographic scale in Kenya and Uganda ([[cite:&nderu-2019-genet-diver]]).

Complementing the DAPC analysis, we performed ADMIXTURE analysis, and found high levels of admixture between all subpopulations of /P. falciparum/. These results can be interpreted in the context of high malaria transmission leading to increased recombination between differnt strains during co-infection. It remains to be investigated, how recombination rates correlate with levels of within-host diversity F_{WS} and genome-wide linkage disequilibrium.

We assigned ancestral states using the outgroup species /P. reichenowi/ and calculated the unfolded site-frequency spectrum (SFS) for chromosomes 2 and 11. The number of SNPs we were able to polarize, was limited to divergent sites between the outgroup and ingroups which were segregating in /P.falciparum/.  While we deem our method of assigning ancestral states to allels to be robust under the infinite-sites model, we acknowledge misorientations could happen due incomplete lineage sorting and sites not fully diverged between the two parasite species or recurrent mutation at the same site (i.e. homplasy) ([[cite:&baudry-2003-effec-misor]]). Including more samples of the outgroup can increase the certainty of selecting sites, that are divergent between /P. reichenowi/ and /P. falciparum/, reducing cross-species polymorphic sites. Additionally, including polarization of alleles using a more distant outgroups, such as /P. vivax/, might yield more sites, that can be assigned anchestral states. We believe that allele misorientations due to sequencing errors of the outgroup are rather unlikely with our method of polarisation, because the allele to be polarized not only should be fixed in the outgroup sequence, but also segregating in the ingroup species. While sequencing errors, that produce the same allele as also found polymorphic in the ingroup are possible, we conclude these to be rather unlikely and sufficient to produce the pervasive u-shaped SFS we observe. We propose investigating measures such as the probability of misorientation used in other studies ([[cite:&baudry-2003-effec-misor]], [[cite:&fay-2000-hitch-under]] ), to assess the impact if has on the SFS.

Another factor, which might affect affect the genetic variation in /Plasmodium/ parasites is their unusually high AT-content, reported to range between 75-85% ([[cite:&videvall-2018-plasm-paras]]). This makes /Plasmodium/ species one of the eukaryotic organisms, with the lowest known genomic GC-content. While the reason for this is still unknown, research has shown, that the common ancestor between /P. falciparum/ and /P. vivax/, was already extremly AT-rich, but recently /P.vivax/ has been gaining GC-content ([[cite:&nikbakht-2014-evolut-genom]]). Such AT bias can be caused by mutational or gene conversion bias, leading to effects of molecular variation not expected by neutral theory. For example, GC- biased gene conversion in bacteria has been shown to skew the site-frequency spectrum and influence demographic inference made from reconstructed genealogies ([[cite:&lapierre-2016-impac-selec]]). While we cannot say if the high AT content is leading to the U-shaped SFS we oberve in /P. falciparum/, further analysis excluding SNPs known for this bias, can help characterize the effect of AT-content. Furthermore, the dynamic and highly variable nature of nucleotide content in closely related species of /Plasmodium/ need to be accounted for when doing comparative analysis of molecular polymorphism.

The classic interpretation of an excess of high-frequency derived variants in the genome involves selective sweeps, driving the increase of these alleles. The U-shaped upper tail in the SFS results from selection at multiple loci, including hitchhiking of neutral variants linked to beneficial alleles, while the lower tail excess of singletons is due to negative selection keeping deleterious mutations at low frequency ([[cite:&bustamante-2001-direc-selec]]). Similar to positive selection, fluctuation selection is also predicted to affect the SFS in a similar way, leading to increased high-frequency derived alleles ([[cite:&huerta-sanchez-2008-popul-genet]]). Comparing the locations of some of these high-frequency SNPs to published candidate genes under selection, can be help identify the contribution of selection to the U-shaped SFS. Additionally, genome-wide recombination rates, can uncover how of genetic hitchhiking affects to levels of diversity in different malaria transmission regions. Because of the parasitic nature of /P. falciparum/, we would expect strong selection to act of genes involved in drug resistance and transmission, as survival of the parasite heavily depends on its ability to infect hosts.

Indeed, several studies have found both directional positive selection and balancing types of selection in /P. falciparum/ (cite:&naung-2022-global-diver,  [[cite:&mobegi-2014-genom-wide]]). Usage of antimalarial drugs, such as chloroquinbe has been shown to drive allele frequency changes over a period of 25 years ([[cite:&nwakanma-2013-chang-malar]]). Interestingly, patterns of selection for drug resistance reflect historical usage of antimalaria drugs varying between countries. Similarly, different regional selection pressures can be expected from geographical differences in transmission intensity and malaria endemicity ([[cite:&mobegi-2014-genom-wide]]), which is reflected in a high divergence (high F_{ST}), between genomic regions between subpopulations. Seasonal differences in transmission intensity can cause selection pressures to fluctuate temporally.

Add paragraph about selection in Plasmodium. Positive directional. High transmission, competitive vs. low transmission non competitive. Drug resistance loci, geographically related to usage of medicines in certain countries. Seasonal fluctuating selection as transmission intensity changes. Balancing selection at host-parasite interface.

Inferring life-history traits... MMC. Furthermore, the SFS can be used to infe




Characterizing genetic variation in /P. falciparum/ is highly important for evalutating the efficacy of control measures and drug interventions, as genetically diverse population have a greater reservoir of alleles, which can selection can act on (*Double check this sentence*). In addition to quantifying the levels of genetic variation, the structure of the parasite population can inform us about the levels of gene flow and outcrossing, which in turn allow alleles to spread between (sub-)populations.

* Discussion of the U-shaped SFS
- gene flow can lead to u-shaped SFS without invoking any directional selection.
- sweepstake reproduction
- misorientation of anchestral and derived alleles. (improve polarisation with other outgroups. Vivax might yield more SNPs, multiple sequences of P.reichenowi to avoid incomplete lineage sorting.)
- biased gene conversion (relevant in Plasmodium, considering extremely high AT content?)
  (might be interesting to calculate the sfs for subset of sites excluding AT bias.)
- directional selection at multiple loci as a cause of u-shaped SFS. And background selection?
- temporally variable selection (makes sense for a pathogen)
- reproductive strategies (i.e. sweepstakes reproduction) and multiple merger events.

* Maybe discussion of MMC models as a more appropriate null model
