#+title: Introduction
#+SETUPFILE: ~/.doom.d/setup-files/latex.setup
#+LATEX_HEADER: \addbibresource{~/biblio/pop-gen.bib}

/Plasmodium falciparum/, an apicomplexan human parasite and the most important disease agent of malaria, is a major global health threat, with nearly half the worlds population at risk for the disease. Especially tropical and subtropical regions suffer the highest disease burden, and nearly 95% of all cases and deaths occur in Africa. Fatalities in this region mainly affect young children, as 80% of malaria mortality occurs in children aged 5 years or younger ([[cite:&world2022world]]). Although impoved control measures have contributed to a steady decline of malaria cases and death rates over the past 20 years, increasing resistances to major anti-malaria drugs, such as artemisin and chloroquinine, are constantly threatening to undermine eradication progress of the disease. Continued effective malaria control measures, rely on disentangling the effects of population structure, selection, recombination and random processes on genome evolution and population dynamics.

Understanding the genetic diversiy and population structure of /P. falciparum/ is crucial to make valid inferences of selection on parasite traits, such as drug resistance. This allows us to seperate adaptive changes in parasite allele frequencies from non-adaptive changes. Importantly, conclusions about selective processes require comprehensive evolutionary 'null'-models, to test hypothesis. Such models should account for population structure, demographic history, as well as other biological or life-history traits, that could potentially lead to signatures of genetic variation similar to selection. Therefore it is important to assess the structure of a sampled parasite population when making further inferences about adaption.

The population genetic structure of /P. falciparum/ has been shown to vary greatly when looking at different global regions, and is closely associated with transmission intensity and disease epidemiology ([[cite:&anderson-2000-micros-marker]]). Regions with low transmission, such as South America or South East Asia, typically show low levels of multiple co-infection, leading to predominantly clonal populations, with low genetic diversity and high levels of linkage disequilibrium (LD). This LD often results from epidemic expansion of a particular clone. On the other hand, high levels of parasite transmission, such as in malaria endemic regions across Africa, leads to high levels of co-infection in the host, high rates of outcrossing and rapid decay in LD ([[cite:&mu-2005-recom-hotsp]]). This variation of population genetic structure makes generalized studies about evolutionary patterns in /P. falciparum/ difficult to conduct and interpret reliably.

Most population genetics inferences are based on the Wright-Fisher model and Kingman coalescent framework to connect the population genetic theory to real-world population genetics data (eg. [[cite:&li-2013-coales-tree]]). In order to assess the genetic variation in a population, summary statistics can be calculated. One such commonly used statistics is the site frequency spectrum (SFS), a histogram of allele frequencies in the population. Neutral theory predicts most allelic variants expected to be rare, because they do not influence an oranisms fitness and will be lost by genetic drift. For a panmictic population, the expected SFS for can be calculated, using coalescent model simulations following the assumptions of Wright-Fisher populations. Typically, the observed SFS spectrum differs from the expected ones, and the type of this skew can reveal details about selective or demographic processes occuring in the population. For example, an excess in rare variants (such as singletons), can be interpreted as signals of positive or negative selection or population expansion.

In nature many SFS are observed to be U-shaped (uSFS) ([[cite:&freund-2023-inter-pervas]]), with an excess of rare and common derived alleles. One common explaination for the uSFS is positive selection at several loci with neutral variants hitchhiking to high-frequencies as well ([[cite:&bustamante-2001-direc-selec]]). This phenoma is predicted to be reinforced by temporally fluctuating selection ([[cite:&huerta-sanchez-2008-popul-genet]]). Many other alternative explaination for a uSFS have been proposed, such as gene flow from non-sampled populations ([[cite:&marchi-2020-gene-flow]]), population subdivision ([[cite:&lapierre-2016-impac-selec]]), biased gene conversion ([[cite:&pouyet-2018-backg-selec]]), misorientation of ancestral and derived alleles ([[cite:&baudry-2003-effec-misor]]) and well as various reproductive strategies ([[cite:&tellier-2014-coales]]).

This great number of factor affecting SFS distributions can make it difficult to effectively identify which forces are dominantly acting on the evolution of an organism. While the Kingman coalescent models can deal with many different assumption such as population structure or changes in population size, it is less suited to deal with large variances in offspring numbers (sweepstake reproduction) and merging of more than two lineages. Sweepstake reproduction applies to species, which produce a high number of offspring together with high early-life mortality, leading to a very recent common ancestor. Similarly, strong and recurrent selective sweeps can lead to the coalescence of multiple branches in the genealogies simultaneously. Because of the limitations the Kingman coalescent is faced with regarding the such life-history traits, species violating these assumptions are better modelled with a more class of coalescent models using multiple merger coalescents (MMC) ([[cite:&tellier-2014-coales]]). Genealogies originating from MMC models, have been observed across all kingdoms of life ([[cite:&freund-2023-inter-pervas]]). Inferences methods based on the classical Wright-Fisher assumptions can lead to erroneous interpretations of different evolutionary mechanisms, if applied to species with strong multiple merger phenomena. Thus for organism with life-cycles that produce MMC genealogies, such as many pathogens or marine organisms, multiple-merger coalescent models have been recommended for inferring the expected SFS ([[cite:&freund-2023-inter-pervas]]).

In this report we analyze population structure and genetic variation of /Plasmodium falciparum/ in select African countries. We calculate the unfolded site-frequency spectrum of the population and provide some context to interpret it in.
