#+title: Introduction
#+SETUPFILE: ~/.doom.d/setup-files/latex.setup
#+LATEX_HEADER: \addbibresource{~/biblio/pop-gen.bib}

/Plasmodium falciparum/, an apicomplexan human parasite and the most important disease agent of malaria, is a major global health threat, with nearly half the worlds population at risk for the disease. Especially tropical and subtropical regions suffer the highest disease burden, with nearly 95% of all cases and deaths occuring in Africa. Fatalities in this region mainly affect young children, as 80% of malaria mortality occurs in children aged 5 years or younger ([[cite:&world2022world]]). Although impoved control measures has contributed to a steady decline of malaria cases and death rates over the past 20 years, increasing resistances to major anti-malaria drugs, such as artemisin and chloroquinine, is constantly threatening to undermine reradication progress of the disease. Continued effective malaria control measures, rely on disentangling the effects of population structure, selection, recombination and random processes on genome evolution and population dynamics.

Understanding the genetic diversiy and population structure of /P. falciparum/ is crucial to make valid inferences of selection on parasite traits, such as drug resistance. This allows us to seperate adaptive changes in parasite allele frequencies from non-adaptive changes. Importantly, conclusions about selective processes require comprehensive evolutionary 'null'-models, to test hypothesis. Such models should account for population structure, demographic history, as well as other biological or life-history traits, that could potentially lead to signatures of genetic variation similar to selection. Therefore it is important to asses the structure of a sampled parasite population when making further inferences about adaption.

Population genetic structure of /P. falciparum/ variy greatly when looking at different global regions, and is closely associated with transmission intensity and disease epidemiology. Regions with low transmission, such as South America or South East Asia, typically show low levels of multiple co-infection, leading to predominantly clonal populations, with low genetic diversity and high levels of linkage disequilibrium (LD). This LD often results from epidemic expansion of a particular clone. On the other hand high levels of parasite transmission, such as in malaria endemic regions across Africa, leads to high levels of co-infection in the host and high rates of outcrossing lead to rapid decay in LD ([[cite:&anderson-2000-micros-marker]], [[cite:&mu-2005-recom-hotsp]]). This variation of population genetic structure makes generalized studies about evolutionary patterns in /P. falciparum/ difficult to conduct and interpret reliably.

In order to assess the genetic variation in a population, sumary statistics can are calculated. One such commonly used statistics is the site frequency spectrum (SFS), a histogram of allele frequencies in the population. Neutral thoery predicts most allelic variants expected to be rare, because they either do not influence an oranisms fitness. For a panmictic population, the expected SFS for neutral sites can be calculated, using coalescent model simulations following the assumptions of Wright-Fisher populations. Typically, the observed SFS spectrum differs from the expected ones, and the type of this skew can reveal details about selective of demographic processes occuring in the population. For example, an excess in rare variants (such as singletons), can be interpreted as signals of purifying selection or population expansion.

In nature many SFS are observed to be U-shaped (uSFS) ([[cite:&freund-2023-inter-pervas]]), with an excess of rare and common derived alleles. One common explaination for the uSFS is positive selection at several loci, due to hitchhiking of neutral variants to high-frequencies together with positively selected variants during selective sweeps ([[cite:&bustamante-2001-direc-selec]]). This phenoma is predicted to be reinforced by temporally fluctuating selection ([[cite:&huerta-sanchez-2008-popul-genet]]).
Many other alternative explaination for a uSFS have been proposed, such as gene flow from non-sampled populations ([[cite:&marchi-2020-gene-flow]]), population subdivision ([[cite:&lapierre-2016-impac-selec]]), biased gene conversion ([[cite:&pouyet-2018-backg-selec]]), misorientation of ancestral and derived alleles ([[cite:&baudry-2003-effec-misor]]) and well as various reproductive strategies ([[cite:&tellier-2014-coales]]). This great number of factor affecting SFS distributions can make it difficult to effectively identify which forces are dominanly acting on the evolution of an organism. Many of the aforementioned factors represent violations to classical Kingman coalescent models, and thus more general multiple-merger coalescent models have been recommended for modeling the expected SFS in many different organisms ([[cite:&freund-2023-inter-pervas]]).

* Additional content? :noexport:
** Importance of accounting for demography
Not only is demography important when accounting for signals of selection, but also relates directly to the epidemiology, as high levels of gene flow and recombination are brought about by high levels of infection and transmission.
- account for demography, when infering life-history traits from coalescent models. Multiple merger models are better?

** Importance of selection

** Questions
- How can demography skew inferences about selection? What happens if we do not account for population structure?
- How does the transmission dynamics and recombination influences genetic variation? What does this mean for control measures and vacciens? What do high levels of gene flow mean for effective control strategies?
- How does the U-shaped sfs influence typical selection inference? Use of multiple merger models?
