#+title: Data and Methods

The analysis described herein aims to explore the genetic diversity, structure of the chosen population. In this section, we provide a comprehensive overview of the data source, data preprocessing steps, and the analytical methods employed for this investigation.
The analysis of population genetics data for /Plasmodium falciparum/ presented in this report was derived from whole-genome Single Nucleotide Polymorphism (SNP) data. The dataset used in this study was obtained from the MalariaGEN Pf7 data resource, a large release of genome variation data from /Plasmodium falciparum/ ([[cite:&MalariaGEN-2023-pf7]]). In whole, Pf7 consists of over 16,000 samples high-quality from over 33 countries and 82 partner studies. By using a method called selective whole genome amplification (sWGA) prior to sequencing, samples could be collected from dried blood spots. While showing the use of sWGA does not indroduce any biases in coverage or population structure, the Pf7 resource is the largest publicly available resource for genomic information of /P.falciparum/. Pf7 provides data on genomic variation as variant calling format (VCF) files per chromosome.
For this report, we chose to analyze 1846 sample from 4 different african countries: Democratic Republik of Congo (DRC), The Gambia, Kenya and Tanzania. The number of samples per country is 520, 452, 285 and 598, respectively and all of them come from 2010-2017, never more than 4 years for any given country. Because of the size of the data, our analysis was conducted for chromosomes 2 and 11, which were downloaded from an ftp server (ftp://ngs.sanger.ac.uk/production/malaria/Resource/34/Pf7_vcf/).
