#+title: Data and Methods

* Pf7 data availability
In this section, we provide a comprehensive overview of the data source, data preprocessing steps, and the analytical methods employed for this investigation.
The analysis of population genetics data for /Plasmodium falciparum/ presented in this report was derived from whole-genome Single Nucleotide Polymorphism (SNP) data. The dataset used in this study was obtained from the MalariaGEN Pf7 data resource, a large release of genome variation data from /Plasmodium falciparum/ ([[cite:&MalariaGEN-2023-pf7]]). In whole, Pf7 consists of over 16,000 samples high-quality from over 33 countries and 82 partner studies. By using a method called selective whole genome amplification (sWGA) prior to sequencing, samples could be collected from dried blood spots. While showing the use of sWGA does not indroduce any biases in coverage or population structure, the Pf7 resource is the largest publicly available resource for genomic information of /P.falciparum/. Pf7 provides data on genomic variation as variant calling format (VCF) files per chromosome. The vcf files are freely available from an FTP server using an FTP client (ftp://ngs.sanger.ac.uk/production/malaria/Resource/34/Pf7_vcf/).

* Data
For this report, we chose to analyze 1846 sample from 4 different african countries: Democratic Republik of Congo (DRC), The Gambia, Kenya and Tanzania. The number of samples per country is 520, 452, 285 and 598, respectively and all of them come from 2010-2017, never more than 4 years for any given country. Because of the size of the data, our analysis was conducted for chromosomes 2 and 11, which we downloaded from the FTP server.

* Filtering and processing
Filtering steps and manipulation of the vcf files were done usingthe software /bcftools/ (*citation*). The VCF files for chromosomes 2 and 11 were filtered to include only biallelic SNPs, which where designated with the filter 'PASS' by the Pf7 data resource. Additionally, we retained only the SNPs with a quality score VQSLOD > 5.0. In total this yielded data sets with 16,269 and 44,040 SNPs, for chromosomes 2 and 11, respectively.
The proportion of heterozygous genotypes in the VCF files was relatively small: 0.2% and 0.3% for chromosome 2 and 11, respectively. To avoid possible effects of multiple infection, we removed all heterozygous genotypes (including both phased and unphased GT fields), by setting them to missing data (./.).


* Analysis
** Population genetics statisics
Basic population genetic measures, such as \theta or Tajima' D statistic, were calculated using the R package /PopGenome/ (*add citation*). Statistics were calculated in sliding windows of 10 Kb for both chromosomes. Additionally, we calculated calculated recombination events accross each chromosome, according to the four-gamete test, also implemented in /PopGenome/.

** Population structure
Local Population structure of the selected data was investigated using discriminant analysis of principal components (dapc), performed using the R package /adegenet/ (*citation*). The dapc analysis was conducted according to the guidelines outlined by [[cite:&thia-2022-guidel-stand]]. To determine the number of clusters suited for discrimination, we performed a K-means clustering algorithm prior to the dapc using /adegenet/'s in-built function /find.clusters()/. The K-means algorithm implemented in /find.cluster()/ uses proir tranformation by principal component analysis (PCA), retaining principal component axis which approximately 80-90% of the variance in the data set. The number of clusters to be used for the dapc was selected based on the Bayesian Information Criterion (BIC). The dapc analysis was run with K-1 PCA axes and K-1 discriminant axes, using the K-mean group assignments as priors.
Extending the dapc analysis, we did a population structure analysis using the software /ADMIXTURE/ (*citation*). The VCF files had their chromosome renamed numerically and using /plink2/ we converted them to .bed files. We conducted admixture analysis for a wide range of K values, including the value used for the dapc. The results of the admixture analysis were validated by comparing the CV error values of each run of K.

** Site-frequency spectrum
In order to calculate an unfolded site-frequency spectrum for the SNP data, we used /Plasmodium reichenowi/ as an outgroup. The genome assembly /PrCDC/ of /P.reichenowi/ was obtained from ([[ftp://ftp.sanger.ac.uk/pub/project/pathogens/reichenowi/2018/October_2018]]). Since the PrCDC assembly was assembled by using the /Plasmodium falciparum/ Pf7 in a refernce guided assembly, we compare the two using genome alignment. We aligned the /P.falciparum/ reference genome (Pf7) with the /P.reichowi/ reference (PrCDC) using /mummer4/ (*citation*), and extracted the segregating sites between both assemblies (*add mapping plot analysis?*).
To polarize the SNPs found in the Pf7 VCF data set, we extract SNPs that overlap with the segregating sites in the PrCDC-Pf7 alignment. If we have segregating sites between our Pf7 samples, the Pf7 reference genome and the PrCDC outgroup they are polarize, if the alternative allele of the VCF file is identical to the allele found on the outgroup sequence. The idea is that the allele found on the outgroup can be considered the anchestral allele, while the allele on the reference must be derived, since it is unlikely that the same allele arose in the Pf7 population and the outgroup.
