#+title: Population Structure and Genetic Variation of /Plasmodium falciparum/ in West and Central Africa
#+subtitle: TUM, Section of population genetics, Prof. Aureli\eacute{}n Tellier \linebreak Research internship final report, WS 2023

#+AUTHOR: Philip L. Wolper
#+EMAIL: phi.wolper@tum.de

#+SETUPFILE: ~/.doom.d/setup-files/bigblow_inline.theme
#+SETUPFILE: ~/.doom.d/setup-files/latex.setup
#+LATEX_HEADER: \addbibresource{~/biblio/pop-gen.bib}

#+begin_src latex
\begin{abstract}
This report presents insights on the genetic diversity and population structure of the malaria-causing parasite /Plasmodium falciparum/ using whole-genome SNP data from various African countries. We confirm previous knowledge of high admixture and limited population substructure between African countries. Our /de novo/ clustering analysis generally confirms this. Interestingly, we find chromosome 11, to exhibit higher signatures of isolation-by-distance than chromosome 2. Additionally, we calculate the unfolded site-frequency spectrum (SFS) for both chromosomes and show it to be U-shaped. We provide some discussion on evolutionary mechanisms leading this U-shaped SFS in /P. falciparum/ and emphasize how population genetics, particularly multiple-merger coalescents, can be applied to understand the evolutionary dynamics of /P. falciparum/ in the future, contributing to better malaria control and drug intervention strategies.
\end{abstract}
#+end_src

# #+LATEX: \vspace{1cm}

* Introduction
#+INCLUDE: "intro.org" :lines "3-"
* Data and Methods
#+INCLUDE: "methods.org" :lines "3-"
* Results
#+INCLUDE: "results.org" :lines "3-"
* Discussion
#+INCLUDE: "discussion.org" :lines "3-"
#+LATEX: \newpage
* Bibliography
[[printbibliography:]]
# bibliography:~/biblio/pop-gen.bib
#+LATEX: \newpage
* Appendix
#+LATEX: \appendix

#+name: admix_Kmeans
#+caption: Admixture analyis of /P. falciparum/ samples, sorted by /de novo/ clusters, identified using K-means.
#+attr_latex: :width 0.9\textwidth :center t :float nil
#+attr_html: :width 500px :align center
[[./figures/admixture_Kmeans_groups.png]]

#+name: SFS_gambia
#+caption: Unfolded site frequency spectrum of Gambian /P. falciparum/ samples.
#+attr_latex: :width 0.9\textwidth :center t :float nil
#+attr_html: :width 600px :align center
[[./figures/sfs_unfolded_chr2_gambian_samples_unlabeled.png]]

#+name: nucmer_coverage_chr2
#+caption: Mapping coverage of /P. reichenowi/ PrCDC aligned to the  /P. falciparum/ Pf3D7 reference genome (/Pf7/).
#+attr_latex: :width 0.6\textwidth :center t : float nil
#+attr_html: :width 400px :align center
[[./figures/Pf7_PrCDC_nucmer_chr2.png]]

#+name: nucmer_coverage_chr11
#+caption: Mapping coverage of /P. reichenowi/ PrCDC aligned to the  /P. falciparum/ Pf3D7 reference genome (/Pf7/).
#+attr_latex: :width 0.6\textwidth :center t : float nil
#+attr_html: :width 400px :align center
[[./figures/Pf7_PrCDC_nucmer_chr11.png]]

** Code used
The complete code used for this report can be found on the git repository: https://github.com/pwolper/Pf7-coevolution

*** Filtering
Downloaded VCF files where filtered and reindexed.
#+begin_src bash
bcftools view \
    --include 'FILTER="PASS" && N_ALT=1 && TYPE="snp" && VSQLOD>5.0'\
    --samples-file $samples_dir/Pf7_african_samples_ids.txt \
    --output-type z\
    --output-file  Pf3D7_02_v3.SNP.vcf \
    Pf3D7_02_v3.pf7.vcf.gz

bcftools index -t Pf3D7_02_v3.SNP.vcf
#+end_src

Heterozygous genotypes were removed using the following code as an example:
#+begin_src bash
bcftools filter -S . -e 'GT=="het"' chr2/Pf3D7_02_v3.afr_samples.qSNP.vcf.gz\
    -o chr2/Pf3D7_02_v3.afr_samples.qSNP.GT_filtered.vcf -Oz
#+end_src

*** DAPC
In order to run the dapc on our vcf files we used the R package /vcfR/ (*citation*) to read in the file and convert it to a genlight object. Subsequently, we read in a vector of country membership and set haploidy. The dapc is conducted according to guideslines stated in [[cite:&thia-2022-guidel-stand]].

#+begin_src R
Pf7.snp <- read.vcfR(vcf_path, verbose = TRUE)

Pf7.snp.gl <- vcfR2genlight(Pf7.snp)

pop(Pf7.snp.gl) <- Pf7.metadata$Country
ploidy(Pf7.snp.gl) <- 1

clust <- find.clusters(Pf7.snp.gl, max.n.clust = 30) #de novo cluster inference
dapc <- dapc(Pf7.snp.gl, n.pca = 3, n.da = 3)
#+end_src
*** ADMIXTURE
Before converting the vcf files were converted to .bed files using /plink2/, we had to rename the chromosome from Pf3D7_x_v3 to x. We used a names.txt file according to the bcftools documentation.

#+begin_src bash
plink2 --vcf ../vcf/chr11/Pf3D7_11_v3.afr_samples.qSNP.GT_filtered.renamed.vcf.gz \
    --make-bed \
    --allow-extra-chr \
    -out Pf3D7_11_v3.afr_samples.qSNP.GT_filtered.renamed

#+end_src

The admixture analysis was performed with the following options:
#+begin_src bash
for K in {2..10}; do
    admixture $bed_file $K --cv | tee log{$K}.out; done
#+end_src

*** Unfolded SFS
The genome sequence of the outgroup /P. reichenowi/ was aligned to the /P. falciparum/ reference sequence Pf7 using the following code:
#+begin_src bash
nucmer --mum -p chr11/Pf7_PrCDC \
    ../Pf7/seqs/Pf3D7_11_v3.fasta ../P.reichenowi/embl.PrCDC/PrCDC_11_v3.fasta

show-coords -c -l -r -T chr11/Pf7_PrCDC.delta > chr11/Pf7_PrCDC_coords.txt

show-snps -T chr11/Pf7_PrCDC.delta > chr11/Pf7_PrCDC_SNPs.txt

awk \
    'NR>3 && $2 !~ /\./ && $3 !~ /\./ {print}' \
    chr11/Pf7_PrCDC_SNPs.txt > chr11/Pf7_PrCDC_11_SNPs_formatted.txt
#+end_src

SNP frequencies and counts were extracted with the following code:
#+begin_src bash
bcftools query -f '%POS %REF %ALT %AF %AC %AN\n'\
    Pf3D7_02_v3.afr_samples.qSNP.GT_filtered.vcf.gz > Pf7.02.vcf.qSNP.AF_AC_AN.txt
#+end_src
