#+title: Population Structure and Gentic Variation of Plasmodium falciparum in West and Central Africa
#+subtitle: Report for a research internship at the Chair of population genetics, Prof. Aureli\eacute{}n Tellier
#+AUTHOR: Philip L. Wolper
#+EMAIL: phi.wolper@tum.de

#+SETUPFILE: ~/.doom.d/setup-files/bigblow_inline.theme
# #+SETUPFILE: ~/.doom.d/setup-files/latex.setup
#+LATEX_HEADER: \addbibresource{~/biblio/pop-gen.bib}
#+OPTIONS: ^:nil p:t

* Introduction
#+INCLUDE: "intro.org" :lines "3-"
* Data and Methods
#+INCLUDE: "methods.org" :lines "3-"
# #+LATEX: \clearpage
* Results
#+INCLUDE: "results.org" :lines "3-"
* Discussion
#+INCLUDE: "discussion.org" :lines "3-"

* Bibliography
[[printbibliography:]]
bibliography:~/biblio/pop-gen.bib

* Appendix
#+LATEX: \appendix

** Figures
#+name: admix_Kmeans
#+caption:
#+attr_latex: :width 1\textwidth :center t
#+attr_html: :width 800px :align center
[[./figures/admixture_Kmeans_groups.png]]

** Code used
*** Filtering
Downloaded VCF files where filtered and reindexed.
#+begin_src bash
bcftools view \
    --include 'FILTER="PASS" && N_ALT=1 && TYPE="snp" && VSQLOD>5.0'\
    --samples-file $samples_dir/Pf7_african_samples_ids.txt \
    --output-type z\
    --output-file  Pf3D7_02_v3.SNP.vcf \
    Pf3D7_02_v3.pf7.vcf.gz

bcftools index -t Pf3D7_02_v3.SNP.vcf
#+end_src

Heterozygous genotypes were removed using the following code as an example:
#+begin_src bash
bcftools filter -S . -e 'GT=="het"' chr2/Pf3D7_02_v3.afr_samples.qSNP.vcf.gz\
    -o chr2/Pf3D7_02_v3.afr_samples.qSNP.GT_filtered.vcf -Oz
#+end_src

*** ADMIXTURE and DAPC
*** Unfolded SFS
