#+title: Results

#+SETUPFILE: ~/.doom.d/setup-files/latex.setup
#+LATEX_HEADER: \addbibresource{~/biblio/pop-gen.bib}

* Diversity Statistics
Across both chromosomes analyzed, we found that diversity statisic, such as \theta_{W}, \theta_{pi}, the number of segregating sites S and Tajima's D to vary quite a lot (Fig. [[div_stats_chr2]] and [[div_stats_chr11]]), as is to be expect from different regions of the genome. Interestingly, we found overall very low values of Tajima's D across all samples (Fig. [[div_stats_chr2]]A and [[div_stats_chr11]]A). When comparing the nucleotide diversity measure of Watterson, \theta_{W} to an estimate of recombination events Hudson-Kaplan-RM, we find a higher diversity for windows with more recombination events (Fig. [[div_stats_chr2]]E and [[div_stats_chr11]]E). This is expected under linked selection, where selection between linked loci will drag down polymorphism. Less linked regions with higher recombination frequencies, therefore are estimated to show more polymorphism. At high recombination rates, this effect, \theta_{W} values flatten out, because polymorphism ultimately is constrained by mutation.

# #+LATEX: \begin{figure}[!htb]
# #+LATEX: \begin{minipage}{0.48\textwidth}

#+name: div_stats_chr2
#+caption: Diversity statistics of chromosome 2, calculated in sliding windows of 10 Kb length. (A) Tajima's D, (B) Segregating sites S, (C) \theta_{Watterson}, (D) \theta_{pi} and (E) \theta_{W} plotted against Hudson-Kaplan RM in 10 Kb windows.
#+attr_latex: :width 0.6\textwidth :float nil
#+attr_html: :width 600px :align center
[[./figures/stats_panel_chr2.png]]

# #+LATEX: \end{minipage}\hfill
# #+LATEX: \begin{minipage}{0.48\textwidth}

#+name: div_stats_chr11
#+caption: Diversity statistics of chromosome 11, calculated in sliding windows of 10 Kb length. (A) Tajima's D, (B) Segregating sites S, (C) \theta_{Watterson}, (D) \theta_{pi} and (E) \theta_{W} plotted against Hudson-Kaplan RM in 10 Kb windows.
#+attr_latex: :width 0.6\textwidth :float nil
#+attr_html: :width 600px :align center
[[./figures/stats_panel_chr11.png]]

# #+LATEX: \end{minipage}
# #+LATEX: \end{figure}

* Population Structure
We conducted two parallel approaches to investigate possible population structure in our /P. falciparum/ samples, discriminant of principle components (dapc) and bayesian structure analysis using the software ADMIXTURE. We first chose to analyze if population structure follows the sampled locations, using these as our a priori labels. Generally we found high levels of admixture between populations, although we could observe some signals of geographic structuring (Fig. [[Pop_Struct_country_chr2]] and [[Pop_Struct_country_chr11]]). Specifically, the Gambian samples showed the most differentiation in the admixture analysis for the other locations. While the distruct plots of te structure analysis show similar low, but present population structure for both chromosomes, it is noteworthy, that chromosome 2 seems to have much less population structure than chromosome 11, as can be seen from the dapc (Fig. [[Pop_Struct_country_chr2]]B) and (Fig. [[Pop_Struct_country_chr11]]B).

#+name: Pop_Struct_country_chr2
#+caption: Population structure analysis of /P. falciparum/ chromosome 2, showing the clustering by predefined labels based on sampling location.
#+attr_latex: :width 1\textwidth :float nil
#+attr_html: :width 700px :align center
[[./figures/Population_structure_countries_chr2.png]]

#+name: Pop_Struct_country_chr11
#+caption: Population structure analysis of /P. falciparum/ chromosome 11, showing the clustering by predefined labels based on sampling location.
#+attr_latex: :width 1\textwidth :float nil
#+attr_html: :width 700px :align center
[[./figures/Population_structure_countries_chr11.png]]

Next, we were interested in possible other causes of population structure in african /P. falciparum/ other than the geographic locations. We inferred the number of groups and membership /de novo/ using a K-means clustering algorithm, /find.clusters()/, described in the dapc manual (*cite*). The optimal value of K was determined to be 9 and 10, respectively for chromosome 2 and 11, significantly more than the a priori groups based on sampling location. We performed dapc analysis on these newly inferred groups, and clustering of both chromosomes into groups, but some of the clusters seem to overlap quite significantly. Chromosome 2 shows more distinct clustering by inferred group labels, than chromosome 11, but they do not correlate with the sampling location or year (Fig. [[dapc_corr_chr2]]). In contrast, the inferred clusters of chromosome 11 seem to  correlate roughly with the sampling location and year. (Fig. [[dapc_corr_chr11]]). When sorting the individuals in the admixture analysis by these inferred group labels, we found the results to agree with the dapc (Fig. [[admix_Kmeans]]).

#+name: dapc_corr_chr2
#+caption: DAPC analysis of nine population clusters inferred /de novo/ with a K-means algorithm, chromosome 2. (A) Individuals colored according to assigned clusters. (B) Individuals colored by their sample location. (C) Individuals are colored by the sampled year.
#+attr_latex: :width 0.8\textwidth :center t
#+attr_html: :width 700px :align center
[[./figures/dapc_Kmeans_chr2.png]]

#+name: dapc_corr_chr11
#+caption: DAPC analysis of nine population clusters inferred /de novo/ with a K-means algorithm, chromosome 2. (A) Individuals colored according to assigned clusters. (B) Individuals colored by their sample location. (C) Individuals are colored by the sampled year.
#+attr_latex: :width 0.8\textwidth :center t
#+attr_html: :width 700px :align center
[[./figures/dapc_Kmeans_chr11.png]]


* Site-frequency spectrum
In order to allow for more detailed inferences about the demographic history of the population and evolutionary forces such as selection we calculated the a polarized site frequency spectrum (SFS), for both chromosomes (Fig. [[SFS_unfolded]]). The SFS we found, both had a U-shaped character, with an excess of singletons compared the the neutral expectation and increased high-frequency alleles. Because hidden population can influence the shape of the SFS, when samples from different populations are pooled together ([[cite:&lapierre-2016-impac-selec]]), we calculated the SFS for samples from Gambia as well, as in chromosome 11 Gambia was the country, with the most distinction from the others. We found the U-shape in the Gambian SFS as well, which indicates the discovered U-shape is not due to unidentified substructuring of the sampled parasites.

#+name: SFS_unfolded
#+caption: Unfolded site frequency for chromosome 2 (A) and 11 (B).
#+attr_latex: :width 0.67\textwidth :center t
#+attr_html: :width 700px :align center
[[./figures/sfs_unfolded.png]]
