#+title: Results

Across both chromosomes analyzed, we found that diversity statisic, such as \theta_{W}, \theta_{pi}, the number of segregating sites S and Tajima's D to vary quite a lot (Fig. [[div_stats_chr2]] and [[div_stats_chr11]]), as is to be expect from different regions of the genome. Interestingly, we found overall very low values of Tajima's D across all samples (Fig. [[div_stats_chr2]]A and [[div_stats_chr11]]A). When comparing the nucleotide diversity measure of Watterson, \theta_{W} to an estimate of recombination events Hudson-Kaplan-RM, we find a higher diversity for windows with more recombination events (Fig. [[div_stats_chr2]]E and [[div_stats_chr11]]E). This is expected under linked selection, where selection between linked loci will drag down polymorphism. Less linked regions with higher recombination frequencies, therefore are estimated to show more polymorphism. At high recombination rates, this effect, \theta_{W} values flatten out, because polymorphism ultimately is constrained by mutation.

#+name: div_stats_chr2
#+caption: Diversity statistics of chromosome 2, calculated in sliding windows of 10 Kb length. (A) Tajima's D, (B) Segregating sites S, (C) \theta_{Watterson}, (D) \theta_{pi} and (E) \theta_{W} plotted against Hudson-Kaplan RM in 10 Kb windows.
#+attr_latex: :width \textwidth :center t
#+attr_html: :width 800px :align center
[[./figures/stats_panel_chr2.png]]

#+name: div_stats_chr11
#+caption: Diversity statistics of chromosome 11, calculated in sliding windows of 10 Kb length. (A) Tajima's D, (B) Segregating sites S, (C) \theta_{Watterson}, (D) \theta_{pi} and (E) \theta_{W} plotted against Hudson-Kaplan RM in 10 Kb windows.
#+attr_latex: :width \textwidth :center t
#+attr_html: :width 800px :align center
[[./figures/stats_panel_chr11.png]]

We conducted two parallel approaches to investigate possible population structure in our /P. falciparum/ samples, dapc analysis and bayesian ADMIXTURE analysis. Both methods are commonly used to identify population structure and complement each other well. We first chose to analyze if population structure follows the sampled locations. Generally we found sign of high levels of admixture between population, although we could observe some signals of geographic structuring. Specifically, the Gambian samples showed the most differntiation in the admixture analysis. It is noteworthy, that chromosome 2 seems to have much less population structure than chromosome 11, with the dapc (Fig. [[Pop_Struct_country_chr2]]B) and (Fig. [[Pop_Struct_country_chr11]]B), supporting this.

#+name: Pop_Struct_country_chr2
#+caption: Population structure analysis of /P. falciparum/ chromosome 2, showing the clustering by predefined labels based on sampling location.
#+attr_latex: :width 1\textwidth :center t
#+attr_html: :width 1000px :align center
[[./figures/Population_structure_countries_chr2.png]]

#+name: Pop_Struct_country_chr11
#+caption: Population structure analysis of /P. falciparum/ chromosome 11, showing the clustering by predefined labels based on sampling location.
#+attr_latex: :width 1\textwidth :center t
#+attr_html: :width 1000px :align center
[[./figures/Population_structure_countries_chr11.png]]
