% Created 2023-06-06 Di 11:07
% Intended LaTeX compiler: pdflatex
\documentclass[11pt]{article}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage{longtable}
\usepackage{wrapfig}
\usepackage{rotating}
\usepackage[normalem]{ulem}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{capt-of}
\usepackage{hyperref}
\usepackage[a4paper,width=160mm,top=25mm,bottom=25mm]{geometry}
\usepackage[style=authoryear]{biblatex}
\usepackage{lmodern} % Ensures we have the right font
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{graphicx}
\usepackage{amsmath, amsthm, amssymb}
\usepackage[table, xcdraw]{xcolor}
\usepackage{float}
\hypersetup{hidelinks}
\definecolor{bblue}{HTML}{0645AD}
\usepackage[explicit]{titlesec}
\setlength{\parindent}{0pt}
\setlength{\parskip}{1em}
\usepackage{titling}
\setlength{\droptitle}{-6em}
\pretitle{\begin{center}\fontsize{20pt}{20pt}\selectfont}
\posttitle{\par\end{center}}
\preauthor{\begin{center}\vspace{-6bp}\fontsize{14pt}{14pt}\selectfont}
\postauthor{\par\end{center}\vspace{-25bp}}
\predate{\begin{center}\fontsize{12pt}{12pt}\selectfont}
\postdate{\par\end{center}\vspace{0em}}
\titlespacing\section{0pt}{5pt}{0pt} % left margin, space before section header, space after section header
\titlespacing\subsection{0pt}{5pt}{0pt} % left margin, space before subsection header, space after subsection header
\titlespacing\subsubsection{0pt}{5pt}{0pt} % left margin, space before subsection header, space after subsection header
\usepackage{enumitem}
\setlist[itemize]{noitemsep, topsep=0pt}
\setlist{itemsep=-2pt} % or
\addbibresource{./biblio/README.bib}
\author{Philip L. Wolper}
\date{\today}
\title{Project Notes for Research Internship in the Lab of Aurelien Tellier\\\medskip
\large Coevolution of human and malaria pathogen under supervision of Meline Saubin.}
\hypersetup{
 pdfauthor={Philip L. Wolper},
 pdftitle={Project Notes for Research Internship in the Lab of Aurelien Tellier},
 pdfkeywords={},
 pdfsubject={},
 pdfcreator={Emacs 27.1 (Org mode 9.6.1)}, 
 pdflang={English}}
\begin{document}

\maketitle


\section{Questions}
\label{sec:org385462e}
\begin{itemize}
\item What does plasmodiums parasitic lifestyle traits contribute it's transmission demography (ie. fluctuating population sizes or mulitple merger events) and adative processes (such as  positive selection and/or coevolution)?
\item Do things like recombination with multiple mergers lead to effects of genetic burden, such as Muller's ratchet or clonal interference?
\item Can we compare \emph{P.falciparum} to /P.vivax?
\item What signatures do the \emph{Pfsa} loci show, when accounting for their demography? What about compared to genes under positive selection?
\end{itemize}

\section{Ideas for analysis}
\label{sec:org9c3313a}
\begin{itemize}
\item Coalescent methods for demogaphic/multiple merger (MMC) methods
\cite{korfmann-2023-simul-infer}
\item SMC? other coalescent methods. Quantifying recombination?
\item Test for balancing selection?
\item Time series of genomic data (ABS) (ask  Meline)
\end{itemize}

Think about the biological question we want to ask!!

\section{HbS-associated variants in \emph{P. falciparum}}
\label{sec:orgfdb52d1}
Described in Band et al are three loci \emph{Pfsa1, 2} and /3/3 (P.falciparum sickle-associated)

\begin{itemize}
\item Pfsa1: chr2: \textbf{631,190 T>A variant}, which lies in ACS8, was associated with HbS. PfACS8 (628,091-632,681)
\item Pfsa2: chr2: \textbf{814,288 C>T variant}, which lies in Pf3D7\_0220300,
\item Pfsa3: chr11: \textbf{1,058,035 T>A variant}, which lies in Pf3D7\_1127000,
\end{itemize}

\textbf{Do these loci show signs of balancing selection indicating coevolution with the host?} What could be the molecular mechanisms and virulence related functions?

\section{Tajima's D [1]}
\label{sec:org183f418}
\begin{center}
\includegraphics[width=.9\linewidth]{./output/TajD/png/Pf7.chr2.full.TajD_DRC_GM_KE.png}
\end{center}

\section{Tajima's D [2]}
\label{sec:orgc2decba}
\begin{center}
\includegraphics[width=.9\linewidth]{./output/TajD/png/Pf7.chr11.full.TajD_DRC_GM_KE.png}
\end{center}

\section{Results?}
\label{sec:org45d3d53}
\subsection{Tajima's D seems to be slightly positive.}
\label{sec:org1fd1620}
Nucelotide diversity:
\begin{itemize}
\item virtual absence of synonymous polymorphisms in genes not invovlved in disease or drug resistance (\cite{rich-2000-popul-struc}).
\begin{itemize}
\item What classifies these genes? Coevolutionary aspects? A high generation of sequence variants? By what mechanisms.
\end{itemize}
\end{itemize}

\section{Signatures of Balancing Selection?}
\label{sec:orgb42e4da}
Signatures of balancing selection:
\begin{itemize}
\item negative-frequency dependent selection
\end{itemize}

What is the gene ontology of genes under balancing selection? How can we show coevolution of these genes? What are some other methods for detecting balancing selection?

Nucelotide diversity:
\begin{itemize}
\item virtual absence of synonymous polymorphisms in genes not invovlved in disease or drug resistance (\cite{rich-2000-popul-struc}).
\begin{itemize}
\item What classifies these genes? Coevolutionary aspects? A high generation of sequence variants? By what mechanisms.
\end{itemize}
\end{itemize}

\section{Pf7 populations}
\label{sec:org2a47976}
\subsection{Kenya}
\label{sec:org99d87bc}
\url{https://www.malariagen.net/apps/pf7/countries/KE}
\begin{itemize}
\item 285 samples from the Year = 2010-2014 \& QC.pass = ``True'' at the locations Kilifi and Kisumu.
\end{itemize}

\subsection{Upper River Gambia}
\label{sec:orgcde26b5}
\url{https://www.malariagen.net/apps/pf7/countries/GM}
\begin{itemize}
\item 452 samples from Upper River and the years 2013-2017
\end{itemize}

\subsection{Democratic Republic of Congo}
\label{sec:orgf4f4067}
\begin{itemize}
\item 520 samples, 2012-2016, QC.pass = True
\end{itemize}

\subsection{Other candidates}
\label{sec:org776838f}
\begin{itemize}
\item Myanmar, Kayin (631 samples, 2016-2017)
\item Malawi, Chikawa (231 samples, 2011)
\item \ldots{}
\end{itemize}
Vivax data set from similar locations?

\section{Data}
\label{sec:orgdf9091d}
vcf files downloaded from: ``\url{ftp://ngs.sanger.ac.uk/production/malaria/Resource/34/Pf7\_vcf/}''

\subsection{SNP filtering with bcf tools.}
\label{sec:orga3336f3}
Filtering for Quality control pass (?), bialleic snps\ldots{}

\begin{verbatim}
bcftools view \
    --include 'FILTER="PASS" && N_ALT=1 && TYPE="snp"'\
    --output-type z\
    --output-file  Pf3D7_02_v3.SNP.vcf \
    Pf3D7_02_v3.pf7.vcf.gz

bcftools index -t Pf3D7_02_v3.SNP.vcf
\end{verbatim}

\subsection{P.vivax data?}
\label{sec:org992a538}
\url{https://wellcomeopenresearch.org/articles/7-136/v1}

\section{Reading list}
\label{sec:org8d0bed9}
Citations and PDF files can be found in the \emph{\textasciitilde{}/biblio} directory.
\subsection{Pf7 populations}
\label{sec:org7cc5de0}
\begin{itemize}
\item \cite{rich-2000-popul-struc}; Rich et al., 2000: Population structure and recent evolution of Plasmodium falciparum
\item \cite{nderu-2019-genet-diver}; Nderu et al., 2019: Genetic diversity and population structure of P. falciparum in Kenyan-Ugandan border areas
\item \cite{amambua-ngwa-2019-major-subpop}; Amambua-Ngwa et al., 2019: Major subpopulations of P. falciparum in sub-Saharan Africa
\item \cite{meyer-2002-review}; Meyer et al., 2002: Genetic diversity of P. falciparum: asexual stages
\end{itemize}

\subsection{Plasmodium Genomics}
\label{sec:org620d310}
\begin{itemize}
\item \cite{benavente-2021-distin-genet}; Benavente, 2021: Genetic structure and selection patterns of Plasmodium vivax in South Asia and East Afrika
\end{itemize}

\subsection{Human-parasite interactions}
\label{sec:org45099fe}
\begin{itemize}
\item \cite{band-2021-malar-protec}; Band, 2021: Malaria Protection due to Sickle Haemoglobin Depends on Parasite Genotype
\item \cite{raberg-2023-human-pathog}; Raberg, 2023: Human and Pathogen Genotype-By-Genotype Interactions in the Light of Coevolution theory
\end{itemize}

\subsection{Coevolution Theory}
\label{sec:org9cc0997}
\begin{itemize}
\item \cite{brown-2011-plant-paras-coevol}; Brown and Tellier, 2011: Plant-parasite coevolution: Bridging the Gap between Genetics and Ecology
\item \cite{tellier-2021-theor-host}; Tellier and Brown, 2021: Theory of Host-Parasite Coevolution: From Ecology to Genomics
\item \cite{maerkle-2021-novel-genom}; Maerkle, 2021: Genomic approaches to study antagonistic coevolution in host and parasites
\end{itemize}

\section{Links}
\label{sec:org145450a}
\subsection{Pf7 Data Viewer}
\label{sec:orgf166341}
\url{https://www.malariagen.net/apps/pf7/}

\subsection{PopGen Cluster - p.wolper home}
\label{sec:orgacbe780}
\url{file://sshx:p.wolper@10.152.154.1:\~/}

\subsection{Kevin Korfmann Kenya Workshop GitHub}
\label{sec:orgacdc2d8}
\url{https://kevinkorfmann.github.io/workshop-kenya/session\_1.html}
\end{document}