#+TITLE: Project Notes for Research Internship in the Lab of Aurelien Tellier
#+date: \today
#+SUBTITLE: Coevolution of human and malaria pathogen under supervision of Meline Saubin.
#+STARTUP: showstars
#+SETUPFILE: report/beamer.setup
#+latex_header: \mode<beamer>{\usetheme{Madrid}}

* Tasks :noexport:
- [ ] Choose individuals from each population for further analysis.
  - how many samples? Closest together around the mean? Coordinated from DAPC. From STRUCTURE analysis.
- [ ] Recombination rate in sliding windows with LDhelmet.
- [ ] STRUCTURE Analysis, then choose 50 'pure' individuals.
  - fastSTRUCTURE
- [X] DAPC and PCA_africa plot
  - running for chr11 as well
  - Maybe with F_st population differentitaion between chooses populations
- [ ] Plot the unfolded SFS.
  - Polarize the vcf file. Call variants from a BAM file obtained using gmap between the Pf7 and PrSy57.
  - Then calculate the unfolded sfs.
  - chr11 and chr2 different types of sites. Sites far away from each other.

* Unfolded sfs
- EasySFS: can calculate the unfolded SFS, but requires accurately polarized vcf (https://github.com/isaacovercast/easySFS)
- SuperSFS: can speculate the anchestral allele, but need the outgroup included in the vcf file (https://github.com/xhchauvet/superSFS)
- repolarize-vcf: can be used to repolarize a vcf file using the genotype of one sample (outgroup) in the vcf. (https://github.com/xuelingyi/repolarize-vcf/blob/main/README.md)

Divergence between falciparum and reichenowi? https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1001099
Reichenowi has been used as an outgroup: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4316729/

* P.reichenowi as an outgroup
The first P. reichenowi reference genome PrCDC was assembled by mapping to Pf7 reference genome ([[cite:&otto-2014-genom-sequen]])
Reference genome PrSY57 has been constructed by mapping to PrCDC ([[cite:&sundararaman-2016-genom-crypt]]).

Check the divergence between PrSY57 anf Pf7. If there is too much intra-specific polymorphism, we cannot determine the ancestral state of the sample polymorphisms.
How to do this? Alignment or mapping?

* Recombination rate
Calculation of recombination rate with FastEPRR not working becasue the data needs to be phased. How can we "phase" the vcf file, when P.falciparum is haploid.
https://academic.oup.com/g3journal/article/6/6/1563/6029955

* PCA and DAPC
https://grunwaldlab.github.io/Population_Genetics_in_R/DAPC.html

* Questions
- What does plasmodiums parasitic lifestyle traits contribute it's transmission demography (ie. fluctuating population sizes or mulitple merger events) and adative processes (such as  positive selection and/or coevolution)?
- Do we find signatures of multiple mergers or dormancy in /P. falciparum/ populations?

- Maybe: What signatures do the /Pfsa/ loci show, when accounting for their demography? What about compared to genes under positive selection?


* Ideas for analysis
- Coalescent methods for demogaphic/multiple merger (MMC) methods [[cite:&korfmann-2023-simul-infer]]


* HbS-associated variants in /P. falciparum/
Described in Band et al are three loci /Pfsa1, 2/ and /3/3 (P.falciparum sickle-associated)

- Pfsa1: chr2: *631,190 T>A variant*, which lies in ACS8, was associated with HbS. PfACS8 (628,091-632,681)
- Pfsa2: chr2: *814,288 C>T variant*, which lies in Pf3D7_0220300,
- Pfsa3: chr11: *1,058,035 T>A variant*, which lies in Pf3D7_1127000,

*Do these loci show signs of balancing selection indicating coevolution with the host?* What could be the molecular mechanisms and virulence related functions?

* Population genetic stats
[[./output/Pf7.chr2.stats.DRC_GM_KE_MM.png]]

** Results
- D_{Taj} seems to be very low (< -2.5) for most regions of chr2 and chr11. Compare with P.vivax (-2) and P.falciparum antigenic genes (1.5-2). Can we confirm these values for chr2 (/Msp4/ , /Sera5/, /Sera8/)?

* PCA plot
#+attr_latex: :placement center :width 1.3\textwidth
[[./output/pca/Pf7.chr2.pca.all.png]]

* Tajima's D :noexport:
** Tajima's D
*** P.falciparum chr 2
[[./output/TajD/png/Pf7.chr2.full.TajD_DRC_GM_KE.png]]

** Tajima's D
*** P.falciparum chr 11
[[./output/TajD/png/Pf7.chr11.full.TajD_DRC_GM_KE.png]]

*  $\theta$ and $\pi$

* Discussion
** Tajima's D peaks?
How can we interpret the values of D_Taj?

- Nucelotide diversity:
- virtual absence of synonymous polymorphisms in genes not invovlved in disease or drug resistance ([[cite:&rich-2000-popul-struc]]).
  - What classifies these genes? Coevolutionary aspects? A high generation of sequence variants? By what mechanisms.

** Signatures of Balancing Selection?
Signatures of balancing selection:
- negative-frequency dependent selection

What is the gene ontology of genes under balancing selection? How can we show coevolution of these genes? What are some other methods for detecting balancing selection?

* Pf7 populations :noexport:
** Kenya
https://www.malariagen.net/apps/pf7/countries/KE
- 285 samples from the Year = 2010-2014 & QC.pass = "True" at the locations Kilifi and Kisumu.

** Upper River Gambia
https://www.malariagen.net/apps/pf7/countries/GM
 - 452 samples from Upper River and the years 2013-2017

** Democratic Republic of Congo
 - 520 samples, 2012-2016, QC.pass = True

** Tanzania
- 589 samples (2010-2014), QC.pass = True

** Other candidates
- Myanmar, Kayin (631 samples, 2016-2017)
- Malawi, Chikawa (231 samples, 2011)
- ...
Vivax data set from similar locations?

* Data
vcf files downloaded from: "ftp://ngs.sanger.ac.uk/production/malaria/Resource/34/Pf7_vcf/"

** SNP filtering with bcf tools.
Filtering for Quality control pass (?), bialleic snps...

#+begin_src bash
bcftools view \
    --include 'FILTER="PASS" && N_ALT=1 && TYPE="snp"'\
    --output-type z\
    --output-file  Pf3D7_02_v3.SNP.vcf \
    Pf3D7_02_v3.pf7.vcf.gz

bcftools index -t Pf3D7_02_v3.SNP.vcf
#+end_src

** P.vivax data?
https://wellcomeopenresearch.org/articles/7-136/v1

* Reading list

Citations and PDF files can be found in the /~/biblio/ directory.

** Plasmodium Genomics
- Rich et al., 2000: Population structure and recent evolution of Plasmodium falciparum; [[cite:&rich-2000-popul-struc]];
- Nderu et al., 2019: Genetic diversity and population structure of P. falciparum in Kenyan-Ugandan border areas; [[cite:&nderu-2019-genet-diver]];
- Amambua-Ngwa et al., 2019: Major subpopulations of P. falciparum in sub-Saharan Africa; [[cite:&amambua-ngwa-2019-major-subpop]];
- Meyer et al., 2002: Genetic diversity of P. falciparum: asexual stages;  [[cite:&meyer-2002-review]];
- Benavente, 2021: Genetic structure and selection patterns of Plasmodium vivax in South Asia and East Afrika; [[cite:&benavente-2021-distin-genet]];
- Nygaard et al, 2010: Long- and Short-term selective forces on Malaria Parasite Genomes;[[cite:&nygaard-2010-long-short]]
- Sundararaman et al. 2016: Genomes of cryptic chimpanzee Plasmodium reveal key evolutionary events leading to human malaria;[[cite:&sundararaman-2016-genom-crypt]]
- Otto et al, 2014: Genome sequencing of chimpanzee malaria parasite reveals possible pathways of adaptation to human hosts; [[cite:&otto-2014-genom-sequen]]
- Pearson et al., 2016: Genomic analysis of local variation and recent evolution in Plasmodium vivax; [[cite:&nygaard-2010-long-short]]
- Naung et al., 2022: Global diversity and balancing selection of 23 leading Plasmodium falciparum candidate vaccine antigens;[[cite:&naung-2022-global-diver]]

** Human-parasite interactions
- Band, 2021: Malaria Protection due to Sickle Haemoglobin Depends on Parasite Genotype; [[cite:&band-2021-malar-protec]];
- Raberg, 2023: Human and Pathogen Genotype-By-Genotype Interactions in the Light of Coevolution theory; [[cite:&raberg-2023-human-pathog]];

** Coevolution Theory
- Brown and Tellier, 2011: Plant-parasite coevolution: Bridging the Gap between Genetics and Ecology; [[cite:&brown-2011-plant-paras-coevol]];
- Tellier and Brown, 2021: Theory of Host-Parasite Coevolution: From Ecology to Genomics; [[cite:&tellier-2021-theor-host]];
- Maerkle, 2021: Genomic approaches to study antagonistic coevolution in host and parasites; [[cite:&maerkle-2021-novel-genom]];

* Links
** Pf7 Data Viewer
https://www.malariagen.net/apps/pf7/

** P.reichenowi ftp
ftp://ftp.sanger.ac.uk/pub/project/pathogens/Plasmodium/reichenowi/

** PopGen Cluster - p.wolper home
[[/sshx:p.wolper@10.152.154.1:~/]]

** Kevin Korfmann Kenya Workshop GitHub
https://kevinkorfmann.github.io/workshop-kenya/session_1.html

