#+TITLE: Project Notes for Research Internship in the Lab of Aurelien Tellier
#+SUBTITLE: Coevolution of human and malaria pathogen under supervision of Meline Saubin.
#+STARTUP: showstars
#+SETUPFILE: ~/.doom.d/latex.setup

* Tasks :noexport:
- [ ] Selection of populations
  - Kenya, Gambia, DRC
  - see Reading List/Pf7 populations, additional populations. Vivax as well?
- [-] Calculate TajD
  - What region size around the locis? More resolved than 500kb?
  - Code Taj-D.R into functions for easy runs.
  - Chr 11; Pfsa3
- [ ] Nucleotide diversity
  - Sliding window around the loci? or whole genome?
  - Using synonymous polymorphisms?
- [ ] F_st population differentitaion between chooses populations
- [ ] Code: switch scripts to be called more as functions. (bcftools, downloading into data.)
  - [ ] sliding window function.

* HbS-associated variants in /P. falciparum/
Described in Band et al are three loci /Pfsa1, 2/ and /3/3 (P.falciparum sickle-associated)

- Pfsa1: chr2: *631,190 T>A variant*, which lies in ACS8, was associated with HbS. PfACS8 (628,091-632,681)
- Pfsa2: chr2: *814,288 C>T variant*, which lies in Pf3D7_0220300,
- Pfsa3: chr11: *1,058,035 T>A variant*, which lies in Pf3D7_1127000,

*Do these loci show signs of balancing selection indicating coevolution with the host?*

* Tajima's D [1]
[[./output/TajD/png/Pf7.chr2.full.TajD_DRC_GM_KE.png]]

* Tajima's D [2]
[[./output/TajD/png/Pf7.chr11.full.TajD_DRC_GM_KE.png]]

* Signatures of Balancing Selection
Signatures of balancing selection:
- negative-frequency dependent selection

What is the gene ontology of genes under balancing selection? How can we show coevolution of these genes? What are some other methods for detecting balancing selection?

Nucelotide diversity:
- virtual absence of synonymous polymorphisms in genes not invovlved in disease or drug resistance ([[cite:&rich-2000-popul-struc]]).
  - What classifies these genes? Coevolutionary aspects? A high generation of sequence variants? By what mechanisms.

* Ideas for analysis
- Coalescent methods for demogaphic/multiple merger (MMC) methods
 [[cite:&korfmann-2023-simul-infer]]

- Time series of genomic data (ABS) (ask  Meline )
- SMC? other coalescent methods.

Think about the biological question we want to ask!!

* Questions
- What does plasmodiums parasitic lifestyle traits contribute it's transmission demography (ie. fluctuating population sizes or mulitple merger events) and adative processes (such as  positive selection and/or coevolution)?
- Can we compare /P.falciparum/ to /P.vivax.

* Pf7 populations
** Kenya
https://www.malariagen.net/apps/pf7/countries/KE
- 285 samples from the Year = 2010-2014 & QC.pass = "True" at the locations Kilifi and Kisumu.

** Upper River Gambia
https://www.malariagen.net/apps/pf7/countries/GM
 - 452 samples from Upper River and the years 2013-2017

** Democratic Republic of Congo
- 520 samples, 2012-2016, QC.pass = True

** Other candidates
- Myanmar, Kayin (631 samples, 2016-2017)
- Malawi, Chikawa (231 samples, 2011)
- ...
Vivax data set from similar locations?

* Data
vcf files downloaded from: "ftp://ngs.sanger.ac.uk/production/malaria/Resource/34/Pf7_vcf/"

** SNP filtering with bcf tools.
Filtering for Quality control pass (?), bialleic snps...

#+begin_src bash
bcftools view \
    --include 'FILTER="PASS" && N_ALT=1 && TYPE="snp"'\
    --output-type z\
    --output-file  Pf3D7_02_v3.SNP.vcf \
    Pf3D7_02_v3.pf7.vcf.gz

bcftools index -t Pf3D7_02_v3.SNP.vcf
#+end_src

** P.vivax data?
https://wellcomeopenresearch.org/articles/7-136/v1

* Reading list
Citations and PDF files can be found in the /~/biblio/ directory.
** Pf7 populations
- [[cite:&rich-2000-popul-struc]]; Rich et al., 2000: Population structure and recent evolution of Plasmodium falciparum
- [[cite:&nderu-2019-genet-diver]]; Nderu et al., 2019: Genetic diversity and population structure of P. falciparum in Kenyan-Ugandan border areas
- [[cite:&amambua-ngwa-2019-major-subpop]]; Amambua-Ngwa et al., 2019: Major subpopulations of P. falciparum in sub-Saharan Africa
- [[cite:&meyer-2002-review]]; Meyer et al., 2002: Genetic diversity of P. falciparum: asexual stages

** Plasmodium Genomics
- [[cite:&benavente-2021-distin-genet]]; Benavente, 2021: Genetic structure and selection patterns of Plasmodium vivax in South Asia and East Afrika

** Human-parasite interactions
- [[cite:&band-2021-malar-protec]]; Band, 2021: Malaria Protection due to Sickle Haemoglobin Depends on Parasite Genotype
- [[cite:&raberg-2023-human-pathog]]; Raberg, 2023: Human and Pathogen Genotype-By-Genotype Interactions in the Light of Coevolution theory

** Coevolution Theory
- [[cite:&brown-2011-plant-paras-coevol]]; Brown and Tellier, 2011: Plant-parasite coevolution: Bridging the Gap between Genetics and Ecology
- [[cite:&tellier-2021-theor-host]]; Tellier and Brown, 2021: Theory of Host-Parasite Coevolution: From Ecology to Genomics
- [[cite:&maerkle-2021-novel-genom]]; Maerkle, 2021: Genomic approaches to study antagonistic coevolution in host and parasites

* Links
** Pf7 Data Viewer
https://www.malariagen.net/apps/pf7/

** PopGen Cluster - p.wolper home
[[/sshx:p.wolper@10.152.154.1:~/]]

** Kevin Korfmann Kenya Workshop GitHub
https://kevinkorfmann.github.io/workshop-kenya/session_1.html

