#+title: Polarizing a vcf with an outgroup sequence.

* Questions
- Does P. reichenowi have (ALT) compared to Pf7_reference. Do the overlap much with the ones from the Pf7_dataset, ie. can we assiggn the true anchestral allele all the time?

* P.reichenowi variant call
To generate a VCF file between the Pf7 reference genome and a single sample from P. reichenowi, you would need the following:

1. Pf7 reference genome FASTA file: This contains the Pf7 reference genome sequence.
2. P. reichenowi sample FASTA file: This contains the assembled genome sequence for the P. reichenowi sample.
3. GFF3 file: This contains the genomic annotations (gene locations, exons, etc.) for the Pf7 reference genome.

Here's a step-by-step guide to generating the VCF file using `bcftools`:

1. Index the reference genome:
#+begin_src bash

   bcftools faidx Pf7_reference.fasta
#+end_src

2. Map P. reichenowi sample to the Pf7 reference genome:
#+begin_src bash
bwa index Pf7_reference.fasta
bwa mem -t <num_threads> Pf7_reference.fasta P_reichenowi_sample.fasta | samtools sort -O BAM -o P_reichenowi_sample.bam
samtools index P_reichenowi_sample.bam
#+end_src

3. Generate the pileup file:
#+begin_src bash
      bcftools mpileup -f Pf7_reference.fasta P_reichenowi_sample.bam -o P_reichenowi_sample.bcf
#+end_src

4. Call variants and generate the VCF file:
#+begin_src bash
   bcftools call -mv -Ov -o P_reichenowi_sample.vcf P_reichenowi_sample.bcf
#+end_src


At this point, you should have the VCF file `P_reichenowi_sample.vcf` containing the variants between Pf7 and the P. reichenowi sample. You can then use this VCF file for downstream analysis or merge it with other samples as required.

* Merging the two vcf files
1. Merge VCF files: If you have separate VCF files for P. falciparum and P. reichenowi, you can merge them using `bcftools merge`. Assuming your P. falciparum VCF file is named "pf.vcf" and the P. reichenowi VCF file is named "pr.vcf", run the following command:

#+begin_src bash
   bcftools merge pf.vcf pr.vcf -Ov -o merged.vcf
#+end_src

   This command will merge the two VCF files into a single VCF file named "merged.vcf".

2. Update sample information: Now, use `bcftools reheader` to update the sample names in the merged VCF file. Suppose you have three P. falciparum samples (F1, F2, F3) and two P. reichenowi samples (R1, R2). Run the following command:

#+begin_src bash
   bcftools reheader -s F1,F2,F3,R1,R2 merged.vcf -o updated.vcf
#+end_src

   This command will update the sample names in the VCF file and create a new file named "updated.vcf".

3. Add genotype data: To add the genotype data for the P. reichenowi samples, use `bcftools annotate`. Assuming your P. reichenowi genotype data is in a separate VCF file named "pr_genotypes.vcf", run the following command:

#+begin_src bash
   bcftools annotate --annotations pr_genotypes.vcf --columns +FORMAT -Ov -o final.vcf updated.vcf
#+end_src

   This command will add the genotype data from "pr_genotypes.vcf" to the "updated.vcf" file and create a new file named "final.vcf".

4. Validate and index the VCF file: Finally, validate the VCF file using `bcftools`, and index it for downstream analysis:

#+begin_src bash
   bcftools view final.vcf > validated.vcf
   bcftools index validated.vcf
#+end_src

   These commands will ensure that the VCF file is valid and create an index file named "validated.vcf.csi" for use with subsequent analysis tools.

Now you can proceed with polarizing the variants using the GitHub tool you mentioned and calculate the unfolded site frequency spectrum (SFS) using the "validated.vcf" file.
